Detections/AzureDevOps/AzDOPatSessionMisuse.yaml

```diff
@@ -0,0 +1,44 @@
+id: ac891683-53c3-4f86-86b4-c361708e2b2b
+name: Azure DevOps Personal Access Token (PAT) misuse
+description: |
+  'Description: This Alert detects whenever a PAT is used in ways that PATs are not normally used.  May require Allowlisting and baselining.
+  Use this query for baselining:
+  AzureDevOpsAuditing
+  | distinct OperationName'
+severity: High
+queryFrequency: 1h
+queryPeriod: 3h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Execution
+  - Impact
+query: |
+
+  // Allowlisted UPNs should likely stay empty
+  let AllowlistedUpns = datatable(UPN:string)[
+  'foo@bar.com'
+  ];
+  // Operation Name parts that will alert
+  let HasAnyBlocklist = datatable(OperationNamePart:string)[
+  'Security.',
+  'Project.',
+  'AuditLog.',
+  'Extension.',
+  ];
+  // Distinct Operation Names that will flag
+  let HasExactBlocklist = datatable(OperationName:string)[
+  'Group.UpdateGroupMembership.Add',
+  'Library.ServiceConnectionExecuted',
+  'Pipelines.PipelineModified',
+  'Release.ReleasePipelineModified',
+  //'Git.RefUpdatePoliciesBypassed'
+  ];
+  AzureDevOpsAuditing
+  | where AuthenticationMechanism startswith "PAT" and
+    (
+      OperationName has_any (HasAnyBlocklist) or
+      OperationName in (HasExactBlocklist)
+    )
+    and ActorUPN !in (AllowlistedUpns)
+  | project TimeGenerated, ProjectName, ActorUPN, ActorDisplayName, IpAddress, UserAgent, OperationName, Details, Data
```

Comment:
Detections require that we output all custom entity types currently available in the UX - timestamp, AccountCustomEntity, HostCustomEntity, IPCustomEntity. In this case:

| extend timestamp = TimeGenerated, AccountCustomEntity = ActorUPN, IPCustomEntity = IPAddress

Reply:
fixed in next commit